### ISSUE TITLE
Small documentation issue in several openstack modules

### ISSUE THREAD

**Issue by parberge** [ansible/ansible#17930]
##### ISSUE TYPE

 - Documentation Report

##### COMPONENT NAME

os_subnet

##### ANSIBLE VERSION

```
ansible 2.2.1.0
```

##### CONFIGURATION

N/A

##### OS / ENVIRONMENT

N/A

##### SUMMARY

Several of the Openstack os_\* modules (e.g http://docs.ansible.com/ansible/os_subnet_module.html) reports two different python versions as requirements:

```
python >= 2.6
python >= 2.7
shade
```

Looking in module `os_subnet.py` I can see that it only has these specified:

```
requirements:
    - "python >= 2.6"
    - "shade"
```

So I guess the requirement `python >= 2.7` is automagically picked up somehow. 
Is the correct way to fix this to remove the item `python >= 2.6` from requirements list?
If so I can send PRs for that.

##### STEPS TO REPRODUCE

N/A

```yaml

```

##### EXPECTED RESULTS

N/A

##### ACTUAL RESULTS

N/A
```

```

**Comment by ansibot** [comment]
@peerster Greetings! Thanks for taking the time to open this issue. In order for the community to handle your issue effectively, we need a bit more information. 

Here are the items we could not find in your description:
- issue type
- ansible version
- component name

Please set the description of this issue with this template:
https://raw.githubusercontent.com/ansible/ansible/devel/.github/ISSUE_TEMPLATE.md

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @emonty @shrews @juliakreger @j2sol @rcarrillocruz @thingee
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @dagnello
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @omgjlk
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by acozine** [comment]
@peerster thanks for reporting this. I see the 2.7 requirement in https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/openstack/os_subnet.py#L105 and I see 2.6 is still listed on https://docs.ansible.com/ansible/latest/modules/os_subnet_module.html. I will investigate.

**Comment by ansibot** [comment]
cc @mnaser
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by mnaser** [comment]
needs_contributor

FWIW: It should by mostly all resolved, however, it looks like the following module is still affected mentioning shade as a requirement:

https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/openstack/os_server_metadata.py

**Comment by ansibot** [comment]
cc @cloudnull @evrardjp @odyssey4me
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by acozine** [comment]
Last reference to `shade` was removed from the `devel` branch by #47359. 

I looked at the `devel` branch and at the two most recent supported releases (`stable-2.7` and `stable-2.6`) - in those three branches, all openstack modules document that `python >= 2.7` is required. 

In Ansible 2.5 (`stable-2.5` branch), we still have references to `shade` and all openstack modules document that `python >= 2.6` is required.

So we have fixed this issue where some modules listed duplicate (and conflicting) python version requirements. And updated some other stuff as well.
